fileshare/event_manager.go

```diff
@@ -125,45 +128,32 @@ func (em *EventManager) DisableNotifications() error {
 	return nil
 }
 
-// EventFunc processes events and handles live transfer state.
-// It should be passed directly to libdrop to be called on events.
-func (em *EventManager) EventFunc(eventJSON string) {
+// OnEvent processes events and handles live transfer state.
+func (em *EventManager) OnEvent(event norddrop.Event) {
 	em.mutex.Lock()
 	defer em.mutex.Unlock()
 
 	if !em.isProd {
-		log.Printf("DROP EVENT: %s", eventJSON)
-	}
-
-	var genericEvent genericEvent
-	err := json.Unmarshal([]byte(eventJSON), &genericEvent)
-	if err != nil {
-		log.Printf("unmarshalling drop event: %s\n%s", err, eventJSON)
-		return
-	}
-
-	switch genericEvent.Type {
-	case requestReceived:
-		em.handleRequestReceivedEvent(genericEvent.Data)
-	case requestQueued:
-	case transferStarted:
-	case transferProgress:
-		em.handleTransferProgressEvent(genericEvent.Data)
-	case transferFinished:
-		em.handleTransferFinishedEvent(genericEvent.Data)
+		log.Printf("DROP EVENT: %s", reflect.TypeOf(event.Kind))
+	}
+
+	switch event.Kind.(type) {
+	case norddrop.EventKindRequestReceived:
+		ev := event.Kind.(norddrop.EventKindRequestReceived)
+		em.handleRequestReceivedEvent(ev)
+	case norddrop.EventKindFileStarted:
+		ev := event.Kind.(norddrop.EventKindFileStarted)
+		em.handleFileStartedEvent(ev)
+	case norddrop.EventKindFileProgress:
```

Comment:
Yes, sorry I forgot to put the comment here. There are much more cases, because currently, we have one big handler for `transferFinished` and inside more events are handled. With new libdrop I have to split it into multiple handlers and I'm adjusting tests first to minimize the regression.

So yes, the transferFinished handler is not used in this PR and that's why we have bunch of "unused xyz" linter errors below